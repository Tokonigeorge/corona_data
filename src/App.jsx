import { useEffect } from "react";
import "./App.css";
import { connect, useDispatch, useSelector } from "react-redux";
import { updateData } from "./store/coronaData/data";

function App() {
  const data = useSelector((state) => state.data);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(updateData({ id: 1 }));
  }, []);

  return (
    <div className="App">
      {console.log(data)}
      <p>Testing</p>
    </div>
  );
}

export default App;
