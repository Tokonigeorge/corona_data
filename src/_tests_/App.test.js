/**
 * @jest-environment jsdom
 */
import React from "react";
import renderer from "react-test-renderer";
import axios from "axios";
import App from "../App";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockData = {
  activeCases: "2000",
  deaths: "2000",
  states: [
    {
      id: 1,
      state: "Lagos",
    },
    {
      id: 2,
      state: "Delta",
    },
  ],
};

jest.mock("axios");

describe("fetchData", () => {
  it("fetches successfully data from an API and updates store", async () => {
    const initialState = {};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    axios.get.mockImplementationOnce(() => Promise.resolve(mockData));
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
    await Promise.resolve();
    //create a snapshot for the component
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("fetches erroneously data from an API", async () => {
    const errorMessage = "Network Error";

    axios.get.mockImplementationOnce(() =>
      Promise.reject(new Error(errorMessage))
    );
  });
});
