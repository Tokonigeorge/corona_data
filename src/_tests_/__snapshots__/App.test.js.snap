// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchData fetches successfully data from an API 1`] = `
<div
  className="App min-h-screen w-screen bg-bg"
>
  <div
    className="w-screen h-screen flex items-center justify-center"
  />
</div>
`;
